test/unit/source_tests.rb

```diff
@@ -99,6 +107,32 @@ class RenderTests < InitTests
       assert_equal exp, subject.partial(@template_name, @file_locals)
     end
 
+    should "only render templates with the matching ext if one is specified" do
+      source = @source_class.new(@root, :ext => 'nm')
+      file_path = Factory.template_file('locals.nm')
+      exp = Nm::Template.new(source, file_path, @file_locals).__data__
+      assert_equal exp, source.render('locals',    @file_locals)
+      assert_equal exp, source.render('locals.nm', @file_locals)
+
+      source = @source_class.new(@root, :ext => 'inem')
+      file_path = Factory.template_file('locals_alt.data.inem')
+      exp = Nm::Template.new(source, file_path, @file_locals).__data__
+      assert_equal exp, source.render('locals',               @file_locals)
+      assert_equal exp, source.render('locals_alt',           @file_locals)
+      assert_equal exp, source.render('locals_alt.data',      @file_locals)
+      assert_equal exp, source.render('locals_alt.data.inem', @file_locals)
+
+      source = @source_class.new(@root, :ext => 'nm')
+      assert_nil source.render('locals_alt',           @file_locals)
+      assert_nil source.render('locals_alt.data',      @file_locals)
+      assert_nil source.render('locals_alt.data.inem', @file_locals)
+
+      source = @source_class.new(@root, :ext => 'data')
+      assert_nil source.render('locals_alt',           @file_locals)
+      assert_nil source.render('locals_alt.data',      @file_locals)
+      assert_nil source.render('locals_alt.data.inem', @file_locals)
```

Comment:
@kellyredding - Yeah, I think that covers the scenarios. The `locals` rendering the `locals_alt` template seems strange on the surface but it makes sense with the glob. I'm cool with this.